modules/rational-persistence.el

```diff
@@ -0,0 +1,35 @@
+;;; rational-persistence.el -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Author: System Crafters Community
+
+;; Commentary
+
+;; Some settings to make emacs more of a continous expierience.
+
+;;; Code:
+
+;; When you visit a file, point goes to the last place it was when file was previously visited.
+(save-place-mode 1)
+
+;; Saving of history (minibuffer, kill-ring, search-ring, keyboard macros, shell commands).
+(setq savehist-save-minibuffer-history t)
+  (setq savehist-additional-variables
+        '(kill-ring
+          search-ring
+          regexp-search-ring
+          last-kbd-macro
+          kmacro-ring
+          shell-command-history))
+
+(savehist-mode)
```

Comment:
`savehist-mode` is also activated in `rational-defaults.el` (see L48-49). You will want to remove it from that module as part of this PR.

There is some additional discussion in Issue #31 which may be relevant as well.

Reply:
Are some of these options also available via setq-default or customize-set-variable?

Reply:
> Are some of these options also available via setq-default or customize-set-variable?

I don't know, i remember recently hearing about the difference of these methods (probably in a System Crafters video) but i don't understand it fully just yet. I think since these variables are not buffer local, `setq` and `setq-default` would do the same thing here? `customize-set-variable` i am not sure at all. 😅

Reply:
The biggest difference with `custom-set-variables` is that it will trigger any associated logic with setting the value. If done correctly by the implementor of the `defcustom`, it does not matter when you set the value (e.g. before/after `require`, before/after starting a mode, etc.). If there is no `:set` on the `defcustom`, it is (almost always) equivalent to using `setq-default`.

Reply:
Thank you! That clears things up!